test/test_message_sync.py

```diff
@@ -47,7 +47,6 @@ def teardown_method(self, _):
         # we need to reenable the garbage collector again
         gc.enable()
 
-    @pytest.mark.timeout(inc(0.2))
```

Comment:
This PR looks good, except these changes to `test/test_message_sync.py`. They were added to prevent endless CI/test runs if deadlocks are present. I'm okay with relaxing them (a lot), but they were there for a purpose. What do you think?

Reply:
isn't pytest-timeout plugin working on CI?

Reply:
I think it is working, why the question? And `pytest.mark.timeout` stems from that plugin.

Reply:
This was a more rethoric question. If pytest-timeout is working then there is no need for adjusting the timeout for the test. endless tests will be cought by the overall timeout